Feat add rate from bracket indice method to rate tax scale like

PiGo86 opened the pull request:
As suggested by @benjello, this PR is intended to allow, for a given tax scale, the determination of the tax rate according to the tax bracket indice.

This feature would be useful for a fix in `openfisca-france` : https://github.com/openfisca/openfisca-france/pull/1831#discussion_r835448750
#### New features

- Introduce `rate_from_bracket_indice` method on `RateTaxScaleLike` class
  - Allows for the determination of the tax rate based on the tax bracket indice

benjello commented:
Thank you very much @PiGo86 for this PR

May I ask you to add to new useful methods very similar to what you already did ?:
-  `rate_from_tax_base`
- `threshold_from_tax_base`

Thanks !

PiGo86 commented:
> Thank you very much @PiGo86 for this PR
> 
> May I ask you to add to new useful methods very similar to what you already did ?:
> 
>     * `rate_from_tax_base`
> 
>     * `threshold_from_tax_base`
> 
> 
> Thanks !

Sure!

For `threshold_from_tax_base`, which threshold should be returned? The lower one or the upper one? If it is the upper one, which value has to be returned for the last bracket?

benjello commented:
The lower one !

PiGo86 commented:
Ok, I have just added the two new methods (`rate_from_tax_base` and `threshold_from_tax_base`).

However, I wonder if it is appropriate to add the methods `rate_from_bracket_indice` and `rate_from_tax_base` to `RateTaxScaleLike` class.

Indeed, I understand that these methods are not really accurate for the `LinearAverageRateTaxScale` class, as the returned rate seems to be false in this context.

It should be OK for `threshold_from_tax_base` method.

In addition, I have moved all tests for these three methods in a new `test_rate_tax_scale_like` test file (it seemed more logical to me). Let me know if this is not convenient.

benjello commented:
@PiGo86 : could you move `rate_from_bracket_indice` and `rate_from_tax_base` to MarginalRateTaxScale ?

Thanks again.

PiGo86 commented:
Ok, done! :wink:
